Data Structures/Trees/stl_map.cpp

```diff
@@ -0,0 +1,70 @@
+#include<iostream>
+//We need to include the set library to use the map and define various operations
+#include<map>
+// We need to include the iterator library which is used to iterate over the key and value of each pair
+//These iterators iterate over the containers(i.e map in this case)
+#include<iterator>
+using namespace std;
+bool checkEmpty(map<int,int> m){
+    if(m.empty())
+        return true;
+    else
+        return false;
+}
+int main(){
+    // A map is container used to store the data in the form of key value pair
+    map <int, int> m;
+    int key , value, n;
+    // insert elements into the map
+    cout << "Enter the number of elements to be inserted into the map" << endl;
+    cin >> n;
+    cout << "Enter the elements to be inserted into the map" << endl;
+    for(int i = 0;i < n;i++){
+        cin >> key >> value;
+        //creating a pair class to assign the elements entered by the user
+        m.insert(pair <int, int> (key, value));
```

Comment:
Can we directly do it? Using the statement: `m[key] = value`. This would be better as it is easier to read.